client/src/app/modules/model-manager/components/omz-import-ribbon-content/omz-import-ribbon-content.component.ts

```diff
@@ -116,6 +149,38 @@ export class OmzImportRibbonContentComponent implements AfterViewInit, OnDestroy
     });
   }
 
+  private _getOptionsByKey(key: keyof ModelDownloaderDTO): string[] {
+    const availableOptions = this.dataSource.data?.reduce((acc, item) => {
+      const fieldValue = item[key];
+      if (Array.isArray(fieldValue)) {
+        acc.push(...fieldValue.flat());
+      } else {
+        acc.push(fieldValue);
+      }
+      return acc;
+    }, []);
+    return Array.from(new Set(availableOptions));
+  }
+
+  get taskTypeOptions(): string[] {
+    return this._getOptionsByKey('task_type');
+  }
+
+  get precisionOptions(): string[] {
+    return this._getOptionsByKey('precision');
+  }
+
+  get frameworkOptions(): string[] {
+    return this._getOptionsByKey('framework');
+  }
```

Comment:
Due to the time complexity and change detection cycles it makes sense to move options calculations out of getters.

Reply:
Fixed